torch_np/_funcs_impl.py

```diff
@@ -1798,3 +1798,23 @@ def min_scalar_type(a: ArrayLike, /):
                 break
 
     return DType(dtype)
+
+
+def pad(array: ArrayLike, pad_width: ArrayLike, mode="constant", **kwargs):
+    if mode != "constant":
+        raise NotImplementedError
+    value = kwargs.get("constant_values", 0)
+    if array.dtype.is_floating_point:
+        typ = float
+    elif array.dtype.is_complex:
+        typ = complex
+    elif array.dtype == torch.bool:
+        typ = bool
+    else:
+        typ = int
```

Comment:
Don't we have a PyTorch dtype -> Python type function?

Reply:
Indeed we didn't. Will extract this into a helper, together with some more cleanup.